client/client.go

```diff
@@ -51,13 +56,20 @@ var (
 	// ErrContextRequired is returned by AcquireWait and AcquireByStateWait when
 	// they are invoked with a nil context.
 	ErrContextRequired = errors.New("context required")
+	// ErrTypeNotFound is returned when the requested resource type (rtype) does not exist.
+	// For this error to be returned, you must set DistinguishNotFoundVsTypeNotFound to true.
+	ErrTypeNotFound = errors.New("resource type not found")
 )
 
 // Client defines the public Boskos client object
 type Client struct {
 	// Dialer is the net.Dialer used to establish connections to the remote
 	// boskos endpoint.
 	Dialer DialerWithRetry
+	// DistinguishNotFoundVsTypeNotFound, if set, will make it possible to distinguish between
+	// ErrNotFound and ErrTypeNotFound. For backwards-compatibility, this flag is off by
+	// default.
+	DistinguishNotFoundVsTypeNotFound bool
```

Comment:
do we need to define a flag somewhere to enable this?

Reply:
Since, by default, we preserve the old behavior, I don't believes that the extra setting via flags is necessary. From the user's point of view, their code would look like:

```
client := boskos.NewClient(...)           # as before
client.DistinguishNotFoundVsTypeNotFound = true
```

Reply:
ack